src/View3D.ts

```diff
@@ -2,217 +2,705 @@
+   * If `false` is given, it will disable AR Quick Look based AR session.
+   * @type {boolean | QuickLookSessionOptions}
+   * @default true
+   */
+  public get quickLook() { return this._quickLook; }
+  /**
+   * Priority array for the AR sessions.
+   * If the two sessions are available in one environment, the session listed earlier will be used first.
+   * If the session name is not included in this priority array, that session will be ignored.
+   * See {@link AR_SESSION_TYPE}
+   * @type {string[]}
+   * @default ["webAR", "sceneViewer", "quickLook"]
+   */
+  public get arPriority() { return this._arPriority; }
+  /**
+   * CSS Selector for the canvas element.
+   * @type {string}
+   * @default "canvas"
+   */
+  public get canvasSelector() { return this._canvasSelector; }
+  /**
+   * Call {@link View3D#init init()} automatically when creating View3D's instance
+   * This option won't work if `src` is not given
+   * @type {boolean}
+   * @default true
+   * @readonly
+   */
+  public get autoInit() { return this._autoInit; }
+  /**
+   * Whether to automatically call {@link View3D#resize resize()} when the canvas element's size is changed
+   * @type {boolean}
+   * @default true
+   */
+  public get autoResize() { return this._autoResize; }
+  /**
+   * Whether to listen {@link https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver ResizeObserver}'s event instead of Window's {@link https://developer.mozilla.org/en-US/docs/Web/API/Window/resize_event resize} event when using the `autoResize` option
+   * @type {boolean}
+   * @default true
+   */
+  public get useResizeObserver() { return this._useResizeObserver; }
+
+  public set skybox(val: View3DOptions["skybox"]) {
+    void this._scene.setSkybox(val);
+    this._skybox = val;
+  }
+
+  public set envmap(val: View3DOptions["envmap"]) {
+    void this._scene.setEnvMap(val);
+    this._envmap = val;
+  }
+
+  public set exposure(val: View3DOptions["exposure"]) {
+    this._renderer.threeRenderer.toneMappingExposure = val;
+    this._exposure = val;
+  }
+
+  public set useGrabCursor(val: View3DOptions["useGrabCursor"]) {
+    this._useGrabCursor = val;
+    this._control.updateCursor();
+  }
+
+  /**
+   * Creates new View3D instance.
+   * @param root A root element or selector of it to initialize View3D
+   * @param {View3DOptions} [options={}] An options object for View3D
+   * @throws {View3DError}
+   * |code|condition|
+   * |---|---|
+   * |{@link ERROR_CODE WRONG_TYPE}|When the root is not either string or HTMLElement|
+   * |{@link ERROR_CODE ELEMENT_NOT_FOUND}|When the element with given CSS selector does not exist|
+   * |{@link ERROR_CODE ELEMENT_NOT_CANVAS}|When the element given is not a \<canvas\> element|
+   * |{@link ERROR_CODE WEBGL_NOT_SUPPORTED}|When the browser does not support WebGL|
+   */
+  public constructor(root: string | HTMLElement, {
+    src = null,
```

Comment:
It may be possible to subtract it as DEFAULT_OPTIONS for cfc later.

Reply:
Sure, but should be done later when we are actually making CFC.